tests/components/alexa/test_capabilities.py

```diff
@@ -279,68 +279,68 @@ async def test_api_select_input(
 
 
 @pytest.mark.parametrize(
-    (
-        "target_activity",
-        "activity_list",
-        "current_activity_index",
-        "target_activity_index",
-    ),
+    ("activity", "activity_list", "target_activity_index"),
     [
-        ("TV", ["TV", "MUSIC", "DVD"], 1, 0),
-        ("MUSIC", ["TV", "MUSIC", "DVD", 1000], 0, 1),
-        ("DVD", ["TV", "MUSIC", "DVD", None], 0, 2),
-        ("BAD DEVICE", ["TV", "MUSIC", "DVD"], 0, None),
-        ("TV", ["TV"], 0, 0),
-        ("BAD DEVICE", [], None, None),
+        ("TV", ["TV", "MUSIC", "DVD"], 0),
+        ("MUSIC", ["TV", "MUSIC", "DVD", 1000], 1),
+        ("DVD", ["TV", "MUSIC", "DVD", None], 2),
+        ("TV", ["TV"], 0),
     ],
 )
 async def test_api_select_activity(
     hass: HomeAssistant,
-    target_activity: str,
+    activity: str,
     activity_list: list[str],
-    current_activity_index: int | None,
     target_activity_index: int | None,
 ) -> None:
     """Test api set activity process."""
-    curr_activty = (
-        activity_list[current_activity_index] if current_activity_index else "None"
-    )
     hass.states.async_set(
         "remote.test",
         "off",
         {
-            "current_activity": curr_activty,
+            "current_activity": activity,
```

Comment:
The previous behavior was to potentially set a different initial state than what is requested later. Now we always set the same initial state as is requested later. Does this not matter for the test?

Reply:
For the test it is not important if the current activity is equal to the target activity, because the function that is being testet does not use the current activity.
But we should maybe think about adding the target/current activity again anyway, so it is not confusing to the reader and to make clear that the test actually does something.

Reply:
If it doesn't matter it's fine like this I think.